scripts/ValidateProjectVersionUpdated.ps1

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+.Synopsis
+    Validates that the package version has been updated
+.Description
+    Validates that the package version has been updated by comparing the version
+    specified in the project file with the latest package version published on
+    NuGet. If the version has not been updated, the script will fail and indicate
+    that the project version neeeds to be updated.
+
+.Parameter projectVersion
+    Specifies the current project version. The project version currently is maintained 
+    in the versioning.props file
+#>
+
+Param(
+	[parameter(Mandatory = $true)]
+	[string]$projectVersion
+)
+
+$packageName = "Microsoft.OpenApi.OData"
+
+# Convert the project semantic version into a number e.g. "1.0.9" -> 109
+[int]$currentProjectVersion = $projectVersion -replace "\."
+
+# API is case-sensitive
+$packageName = $packageName.ToLower()
+$url = "https://api.nuget.org/v3/registration5-semver1/$packageName/index.json"
+
+# Call the NuGet API for the package and get the current published version.
+$nugetIndex = Invoke-RestMethod -Uri $url -Method Get
+$currentPublishedVersion = $nugetIndex.items[0].upper
+
+# Convert the published version into a number e.g. "1.0.9" -> 109
+[int]$currentPublishedVersionNumber = $currentPublishedVersion -replace "\."
```

Comment:
the comparison needs to be member by member.
having a published version of 1.0.100 and a project version of 1.2.0 will fail that test although it is correct

Reply:
Okay - I am updating version comparison to check member by member.

Reply:
one way you could make it easier for you would be to rely on the version class from dotnet https://docs.microsoft.com/en-us/dotnet/api/system.version?view=net-6.0 (you can load dotnet classes in PS scripts)

Reply:
Good suggestion. The script has been updated to compare the version using `System.Version`